pkg/server/startup.go

```diff
@@ -58,12 +60,36 @@ func Startup(ctx context.Context, systemNamespace, kubeConfig string) error {
 
 	ctx, rioContext := types.BuildContext(ctx, systemNamespace, restConfig)
 
+	webhookPort := ":443"
+	if _, err := rioContext.Apps.Apps().V1().Deployment().Get(systemNamespace, "rio-controller", metav1.GetOptions{}); errors.IsNotFound(err) {
+		constants.DevMode = "true"
+		controllerStack := stack.NewSystemStack(rioContext.Apply, systemNamespace, "rio-controller")
+		answer := map[string]string{
+			"NAMESPACE": systemNamespace,
+		}
+		if err := controllerStack.Deploy(answer); err != nil {
+			return err
+		}
+		webhookPort = ":7443"
+	}
```

Comment:
So this block checks to see if rio-controller exists, and if it doesn't we know it's dev mode, but then if thats true it deploys rio-controller anyway ? Doesn't that conflict with the local dev controller ?

Reply:
In this rio-controller only if you send `RUN_CONTROLLER=true` as an answer so that it can deploy rio-controller. https://github.com/rancher/rio/pull/548/files/25d50f0ca2e14a5c63907169c206d0c469922698#diff-fc011507dab2bf2f5d8b5041a7a4d91cR278

Reply:
Ok, I think this will be solved by our feature flag convo.